### ISSUE TITLE
False positives for multi-line selector lists containing functional pseudo-classes in indentation

### ISSUE THREAD

**Issue by EliRobinson** [stylelint/stylelint#1741]
A bug, when I use the code:

```
.foo input:not([type=checkbox]):not([type=radio]),
.foo textarea {
```

Stylelint complains that line 2 `.foo textarea {` needs to be indented

> Which rule, if any, is this issue related to?

`indentation`

> What CSS is needed to reproduce this issue?

e.g.

``` css
.foo input:not('[type=checkbox]'):not('[type=radio]'),
.foo textarea {
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

``` json
{
  "rules": {
    "indentation": 4,
  }
}
```

> Which version of stylelint are you using?

`6.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

With `grunt-stylelint`
`stylelint: {
            development: {
                options: {
                    //format: 'less'
                    syntax: 'less'
                },
                src: 'src/public/less/**/*.less'
            }
        },`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Uses Less and includes some :nots followed by a next line selector

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```
>>  39:1  ✖  Expected indentation of 4 spaces   indentation
```

**Comment by evilebottnawi** [comment]
@EliRobinson it is valid css `.foo input:not('[type=checkbox]'):not('[type=radio]'),`?

**Comment by evilebottnawi** [comment]
@EliRobinson https://jigsaw.w3.org/css-validator/validator 
```shell
К сожалению, мы обнаружили следующие ошибки (1)
URI : TextArea
1		Ошибка разбора [:not('[type=checkbox]'):not('[type=radio]')]
```

**Comment by jeddy3** [comment]
I think this is being caused by [this reported issue](https://github.com/webschik/postcss-less/issues/56) within the *experimental* `postcss-less` parser.

The following code also triggers the warning.

```css
.foo input:not(.a),
.foo textarea {}
```

@EliRobinson If you'd like to see this issue resolved, please consider helping with `postcss-less` and addressing the problem there.

**Comment by EliRobinson** [comment]
@evilebottnawi - You're right, I'll double check my code, there shouldn't be a single quote in the `:not` statements.

**Comment by EliRobinson** [comment]
@jeddy3 - Good catch, I thought I searched for the right things to find dupes, apparently not. I fully understood that it was experimental, but still a bug either way. 😄

**Comment by EliRobinson** [comment]
@evilebottnawi - I did have that wrong (though, the browser correctly handled it, weird), however, the issue I mentioned above is still broken. I'll update my original comment to match valid code.

**Comment by baivong** [comment]
My code has the same warning.

```css
.foo li:nth-child(5),
.bar {
```

**Comment by evilebottnawi** [comment]
@baivong `status: blocked` bug in `postcss-less`, you can help their to resolve this issue

**Comment by Ricozor** [comment]
Hi @evilebottnawi,
I have the same warning on a project but I don't use postcss-less in grunt
I always have the bug if I use ':'... (:not(), :first-child,...)
but if i write on one line `.foo li:nth-child(5), .bar {` I have no warning

**Comment by evilebottnawi** [comment]
@Ricozor bug in `postcss-less` Need resolve https://github.com/webschik/postcss-less/issues/60 and https://github.com/stylelint/stylelint/pull/1899 (i can help with this issue). You can help to resolve this issue and i am fixes this in `stylelint`.

**Comment by jwilsson** [comment]
I just tested this on the `v8` branch and it's working properly now (i.e. no warnings were reported).

**Comment by jeddy3** [comment]
Fantastic!
